src/test/java/org/jenkinsci/plugins/build_token_root/BuildRootActionTest.java

```diff
@@ -112,6 +114,28 @@ private void assertCreated(Page page) throws Exception {
         j.waitUntilNoActivity();
     }
 
+    @Test public void responseStatus() throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.getBuildersList().add(new Shell("sleep 2"));
```

Comment:
There are various ways to replace this. The most straightforward I can think of offhand is to switch this to a Pipeline test. Then add the `tests` JAR (`classifier` in `dependency`) of the `workflow-support` plugin and you can use the test-only `semaphore` step to pause. From the test body you would use two `SemaphoreStep` methods to wait for the build to enter that state, and then to allow it to resume. Basically similar to the user-facing `input` step but designed expressly for the needs of `JenkinsRule`-based tests. You can find dozens of examples in various `workflow-*-plugin` repos.

This test would be a bit special in that you would explicitly need to set the `WorkflowJob` to disallow concurrent builds. (By default Pipeline allows concurrent builds, which would bypass the problem needing a fix.)

Reply:
Thanks for the directions, worked like a charm!